File: week06/15/4_Thursday_reverse_bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        lst = []
        for i in range(32):
            lst.append(n%2)
            n >>= 1

        ans = 0
        for i in range(32):            
            ans <<= 1
            ans += lst[i]
        
        return ans

File: week06/15/4_Thursday_reverse_bits.c
#include <stdint.h>

uint32_t reverseBits(uint32_t n) {
    unsigned int ans = 0;
    for (int i = 0; i < 32; i++) {
        ans = (ans << 1) + (n & 1);
        n >>= 1;
    }
    return ans;
}